pkg/controller/identity/iamgroup/controller.go

```diff
@@ -0,0 +1,197 @@
+	errSDK              = "empty IAM Group received from IAM API"
+
+	errKubeUpdateFailed = "cannot late initialize IAM Group"
+)
+
+// SetupIAMGroup adds a controller that reconciles Groups.
+func SetupIAMGroup(mgr ctrl.Manager, l logging.Logger) error {
+	name := managed.ControllerName(v1alpha1.IAMGroupGroupKind)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha1.IAMGroup{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha1.IAMGroupGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), newClientFn: iam.NewGroupClient}),
+			managed.WithConnectionPublishers(),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type connector struct {
+	kube        client.Client
+	newClientFn func(ctx context.Context, credentials []byte, region string, auth awsclients.AuthMethod) (iam.GroupClient, error)
+}
+
+func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mg.(*v1alpha1.IAMGroup)
+	if !ok {
+		return nil, errors.New(errNotGroupInstance)
+	}
+
+	p := &awsv1alpha3.Provider{}
+	if err := c.kube.Get(ctx, meta.NamespacedNameOf(cr.Spec.ProviderReference), p); err != nil {
+		return nil, errors.Wrap(err, errGetProvider)
+	}
+
+	if aws.BoolValue(p.Spec.UseServiceAccount) {
+		groupClient, err := c.newClientFn(ctx, []byte{}, p.Spec.Region, awsclients.UsePodServiceAccount)
+		return &external{client: groupClient, kube: c.kube}, errors.Wrap(err, errCreateGroupClient)
+	}
+
+	if p.GetCredentialsSecretReference() == nil {
+		return nil, errors.New(errGetProviderSecret)
+	}
+
+	s := &corev1.Secret{}
+	n := types.NamespacedName{Namespace: p.Spec.CredentialsSecretRef.Namespace, Name: p.Spec.CredentialsSecretRef.Name}
+	if err := c.kube.Get(ctx, n, s); err != nil {
+		return nil, errors.Wrap(err, errGetProviderSecret)
+	}
+
+	groupClient, err := c.newClientFn(ctx, s.Data[p.Spec.CredentialsSecretRef.Key], p.Spec.Region, awsclients.UseProviderSecret)
+	return &external{client: groupClient, kube: c.kube}, errors.Wrap(err, errCreateGroupClient)
+}
+
+type external struct {
+	kube   client.Client
+	client iam.GroupClient
+}
+
+func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mgd.(*v1alpha1.IAMGroup)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
+	}
+
+	observed, err := e.client.GetGroupRequest(&awsiam.GetGroupInput{
+		GroupName: aws.String(meta.GetExternalName(cr)),
+	}).Send(ctx)
+
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(iam.IsErrorNotFound, err), errGet)
+	}
+
+	if observed.Group == nil {
+		return managed.ExternalObservation{}, errors.New(errSDK)
+	}
+
+	group := *observed.Group
+	current := cr.Spec.ForProvider.DeepCopy()
+	iam.LateInitializeGroup(&cr.Spec.ForProvider, &group)
```

Comment:
There's just one field that can be updated, I think we don't nee a method for that.